same-optype/script.whizzml

```diff
@@ -0,0 +1,28 @@
+;; Given a source with incorrect optypes and a dataset with correct
+;; optypes, returns a dataset built from the source which has the same
+;; field optypes as the given dataset, matching field names. It also
+;; has the same field locales and term_analysis, if applicable.
+
+(define (same-optype source-id dataset-id)
+  (let (ds (fetch dataset-id)
+        ds-ids (keys (ds "fields"))
+        ds-locale (ds "locale")
+        src-fields (resource-fields source-id)
+        src-ids (keys src-fields)
+        ds-fields (map (lambda (x) 
+                         (find-field (ds "fields") 
+                                     (get-in src-fields [x "name"]))) src-ids)
```

Comment:
no need for `get-in`, `(src-fields [x "name"])` should do the same.  also, better put `src-ids` in the next line, nicely aligned with `(lambda`.  are we considering the possibility of a field in the source that is not in the dataset? perhaps it was excluded when creating the dataset... that should amount to use only those ids in the dataset's `input_fields`.